pageserver/src/consumption_metrics.rs

```diff
@@ -83,10 +86,18 @@ pub async fn collect_metrics(
                 return Ok(());
             },
             _ = ticker.tick() => {
-                if let Err(err) = collect_metrics_iteration(&client, &mut cached_metrics, metric_collection_endpoint, node_id, &ctx).await
+                // send cached metrics every SEND_CACHED_ITER_STEP iterations
+                iter_num += 1;
+                let send_cached = iter_num % SEND_CACHED_ITER_STEP == 0;
```

Comment:
This could also be:
```suggestion
                let send_cached = iter_num >= SEND_CACHED_ITER_STEP;
```

Since you reset it (though would move the reset closer to this one from lines 97...100). Whenever I see a `%` used for this I need to think extra-carefully does this only allow on 0 or end or both. I think the original would send on 10th, reset to 0, then send on 10th again because of the `iter_num += 1`.

Is this meant to clear out even on error..? It might be safer that way.

Reply:
Thank you. I used this approach in new version